header.php

```diff
@@ -85,15 +85,21 @@
 
 							<?php else : ?>
 
-								<a href="#menu" id="mobile-menu-button">
-									<?php siteorigin_north_display_icon( 'menu' ) ?>
-									<?php if ( siteorigin_setting( 'responsive_menu_text' ) ) : ?>
-										<?php echo esc_html( siteorigin_setting( 'responsive_menu_text' ) ); ?>
-										<span class="screen-reader-text"><?php esc_html_e( 'Menu', 'siteorigin-north' ); ?></span>
-									<?php endif; ?>
-								</a>
+								<?php
+								$menu_id = ( wp_get_nav_menu_name( 'primary' ) ? wp_get_nav_menu_name( 'primary' ) :  false );
+								if ( wp_get_nav_menu_object( $menu_id )->count || siteorigin_setting( 'navigation_search' ) ) : ?>
```

Comment:
If no menu is assigned to the primary menu location a notice is returned: `Notice: Trying to get property 'count' of non-object in /Users/andrew/Dropbox/Sites/siteorigin/wp-content/themes/siteorigin-north/header.php on line 90`.